numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1788,6 +1791,35 @@ unpack_bits(PyObject *input, int axis, PyObject *count_obj, char order)
         goto fail;
     }
 
+    /* setup lookup table under GIL, big endian 0..256 as bytes */
+    if (unpack_init == 0) {
+        npy_uint64 j;
+        for (j=0; j < 256; j++) {
+            npy_uint64 v_b = 0;
+            npy_uint64 v_l;
+            v_b |= (npy_uint64)((j &   1) ==   1);
+            v_b |= (npy_uint64)((j &   2) ==   2) << 8;
+            v_b |= (npy_uint64)((j &   4) ==   4) << 16;
+            v_b |= (npy_uint64)((j &   8) ==   8) << 24;
+            v_b |= (npy_uint64)((j &  16) ==  16) << 32;
+            v_b |= (npy_uint64)((j &  32) ==  32) << 40;
+            v_b |= (npy_uint64)((j &  64) ==  64) << 48;
+            v_b |= (npy_uint64)((j & 128) == 128) << 56;
+
+            /* for bitorder little the lookup table is just byte swapped */
+            v_l = npy_bswap8(v_b);
+
+            /* the table is stored as big endian, swap the native integer */
```

Comment:
it is probably a little odd that we store the table in big endian as the most common platform is little endian
it was like that originaly as the packbit worked in "big endian" bits, now that it can do both should we change the lookup table to be stored in little endian?

Reply:
+1 from me

Reply:
Wait, isn't the lookup table already little endian, with the low-order bits at the lowest byte of `v_b`?

Reply:
little and big endian are a bit confusing in this context, what is relevant is that the byte pattern of the lookup table is the _same_ on both endianess platforms.
endianess is only relevant when interpreting the bytes as integer which is not done by this function it only extends a byte to 8 bytes and integer is simply the datatype used to construct the bytes.
I'll change the comments to say fixed endian.